src/inmanta/deploy/scheduler.py

```diff
@@ -250,14 +244,11 @@ async def read_version(
         - schedules any resources that are not in a known good state.
         - rearranges deploy tasks by requires if required
         """
-        environment = await data.Environment.get_by_id(self.environment)
-        if environment is None:
-            raise ValueError(f"No environment found with this id: `{self.environment}`")
-        # FIXME[8119]: version does not necessarily correspond to resources' version
-        #       + last_version is reserved, not necessarily released
-        #       -> call ConfigurationModel.get_version_nr_latest_version() instead?
-        version = environment.last_version
-        resources_from_db = await self._build_resource_mappings_from_db()
+        cm_version = await ConfigurationModel.get_latest_version(self.environment)
+        if cm_version is None:
+            return
+        version = cm_version.version
+        resources_from_db = await self._build_resource_mappings_from_db(version=version)
```

Comment:
@sanderr this was the most elegant way I could come up with.

It guarantees consistency and it works for versions with no resources

Reply:
I like it